src/account_id_ref.rs

```diff
@@ -229,6 +229,26 @@ impl AccountIdRef {
     pub const fn len(&self) -> usize {
         self.0.len()
     }
+
+    /// Returns parent's account id reference
+    /// 
+    /// ## Examples
+    /// ```
+    /// use near_account_id::AccountIdRef;
+    /// 
+    /// let alice: &AccountIdRef = AccountIdRef::new("alice.near").unwrap();
+    /// let parent: &AccountIdRef = alice.get_parent_account_id().unwrap();
+    /// 
+    /// assert!(alice.is_sub_account_of(parent));
+    /// 
+    /// let near: &AccountIdRef = AccountIdRef::new("near").unwrap();
+    /// 
+    /// assert!(near.get_parent_account_id().is_none());
+    /// ```
+    pub fn get_parent_account_id(&self) -> Option<&AccountIdRef> {
+        let parent_str = self.as_str().split_once('.')?.1;
+        AccountIdRef::new(parent_str).ok()
```

Comment:
Actually, since we know that the parent account id is always a valid account, we can use `new_unvalidated` here to avoid unnecessary validation.

```suggestion
        AccountIdRef::new_unvalidated(parent_str)
```